src/Og.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\og\Og.
+ */
+
+namespace Drupal\og;
+
+use Drupal\Core\Config\Entity\ConfigEntityInterface;
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\Core\Entity\EntityInterface;
+
+/**
+ * A static helper class for OG.
+ */
+class Og {
+
+  /**
+   * @var string
+   */
+  const MODULE_CONFIG_KEY = 'og';
+
+  /**
+   * @var string
+   */
+  const GROUP_CONFIG_KEY = 'group';
+
+  /**
+   * A static cache of entity type group mappings.
+   *
+   * @var array
+   */
+  protected static $groupEntityCache = [];
+
+  /**
+   * Check if the given entity is a group.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity object.
+   *
+   * @return bool
+   *   True or false if the given entity is group.
+   */
+  public static function isGroup(EntityInterface $entity) {
+    $entity_type_id = $entity->getEntityTypeId();
+    $entity_bundle = $entity->bundle();
+
+    if (isset(static::$groupEntityCache[$entity_type_id][$entity_bundle])) {
+      return static::$groupEntityCache[$entity_type_id][$entity_bundle];
+    }
+
+    if ($entity instanceof ContentEntityInterface) {
+      $entity_type = $entity->getEntityType();
+      $bundle_key = $entity_type->getKey('bundle');
+      $bundle_entity = $entity->get($bundle_key)->entity;
+
+      // @todo Or ...
+//      $bundle_entity_type = $entity_type->getBundleEntityType();
+//      /** @var \Drupal\Core\Config\Entity\ConfigEntityInterface $bundle_entity */
+//      $bundle_entity = \Drupal::entityManager()->getStorage($bundle_entity_type)->load($entity->bundle());
+
+      $is_group = static::isGroupEntityType($bundle_entity);
+      static::$groupEntityCache[$entity_type_id][$entity_bundle] = $is_group;
+
+      return $is_group;
+    }
+
+    static::$groupEntityCache[$entity_type_id][$entity_bundle] = FALSE;
+    return FALSE;
+  }
+
+  /**
+   * Returns whether an entity type instance is a group type.
+   *
+   * @param \Drupal\Core\Config\Entity\ConfigEntityInterface $entity_type
+   *
+   * @return bool
+   */
+  public static function isGroupEntityType(ConfigEntityInterface $entity_type) {
```

Comment:
Yes, that list sounds right
On Sep 19, 2015 12:10 PM, "Damian Lee" <notifications@github.com> wrote:

> In src/Og.php <https://github.com/amitaibu/og/pull/2#discussion_r39918186>
> :
>
> > +
> > +      return $is_group;
> > +    }
> > +
> > +    static::$groupEntityCache[$entity_type_id][$entity_bundle] = FALSE;
> > +    return FALSE;
> > +  }
> > +
> > +  /**
> > +   * Returns whether an entity type instance is a group type.
> > +   *
> > +   * @param \Drupal\Core\Config\Entity\ConfigEntityInterface $entity_type
> > +   *
> > +   * @return bool
> > +   */
> > +  public static function isGroupEntityType(ConfigEntityInterface $entity_type) {
>
> So we just have some base OG config, and we add a list of entity type and
> bundles there, just like:
>
> entity_type:
>   - bundle
>   - bundleother_entity_type:
>   - bundle
>
> Thoughts?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/amitaibu/og/pull/2/files#r39918186>.
>